homeassistant/components/sector_alarm/__init__.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_CODE, CONF_EMAIL
+from homeassistant.helpers import discovery
+
+DOMAIN = 'sector_alarm'
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['sectoralarmlib==0.7']
+
+CONF_ALARM_ID = 'alarm_id'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN:
+    vol.Schema(
+        {
+            vol.Required(CONF_EMAIL): cv.string,
+            vol.Required(CONF_PASSWORD): cv.string,
+            vol.Required(CONF_ALARM_ID): cv.string,
+            vol.Required(CONF_CODE): cv.string
+        }),
+},
+                           extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Initialitation for Sector Alarm."""
+    try:
+        import sectoralarmlib.sector as sectorlib
+
+        alarm = sectorlib.SectorAlarm(config[DOMAIN][CONF_EMAIL],
+                                      config[DOMAIN][CONF_PASSWORD],
+                                      config[DOMAIN][CONF_ALARM_ID],
+                                      config[DOMAIN][CONF_CODE])
+    except RuntimeError:
```

Comment:
There is no call to login here to warrant a try/except

Reply:
We should login here and catch the error. On error we should log error and fail set up by returning `False`.

It is preferable if the library would raise its own custom exceptions, with a common base exception class specific to the library. We can then import those exceptions as needed and use here.